sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/implementation/encryption/DataEncryptionKeyContainerCore.java

```diff
@@ -96,39 +99,42 @@ public DataEncryptionKeyContainerCore(CosmosDataEncryptionKeyProvider dekProvide
                 DataEncryptionKeyProperties dekProperties = result.getT1();
                 InMemoryRawDek inMemoryRawDek = result.getT2();
 
-                Tuple3<byte[], EncryptionKeyWrapMetadata, InMemoryRawDek> wrapResult =
-                    this.wrapAsync(
-                        id,
-                        inMemoryRawDek.getDataEncryptionKey().getRawKey(),
-                        dekProperties.encryptionAlgorithm,
-                        newWrapMetadata);
-
-                byte[] wrappedDek = wrapResult.getLeft();
-                EncryptionKeyWrapMetadata updatedMetadata = wrapResult.getMiddle();
-                InMemoryRawDek updatedRawDek = wrapResult.getRight();
-
-                CosmosItemRequestOptions effectiveRequestOptions = requestOptions != null ? requestOptions : new CosmosItemRequestOptions();
-
-                effectiveRequestOptions.setIfMatchETag(dekProperties.eTag);
-
-                DataEncryptionKeyProperties newDekProperties = new DataEncryptionKeyProperties(dekProperties);
-                newDekProperties.wrappedDataEncryptionKey = wrappedDek;
-                newDekProperties.encryptionKeyWrapMetadata = updatedMetadata;
-
-                Mono<CosmosItemResponse<DataEncryptionKeyProperties>> responseMono = this.dekProvider.getContainer().replaceItem(
-                    newDekProperties,
-                    newDekProperties.id,
-                    new PartitionKey(newDekProperties.id),
-                    effectiveRequestOptions);
-
-                return responseMono.flatMap(
-                    response -> {
-                        DataEncryptionKeyProperties item = response.getItem();
-
-                        assert (item != null);
-                        this.dekProvider.getDekCache().setDekProperties(id, item);
-                        this.dekProvider.getDekCache().setRawDek(id, updatedRawDek);
-                        return Mono.just(response);
+                Mono<Tuple3<byte[], EncryptionKeyWrapMetadata, InMemoryRawDek>> wrapResultMono = this.wrapAsync(
+                    id,
+                    inMemoryRawDek.getDataEncryptionKey().getRawKey(),
+                    dekProperties.encryptionAlgorithm,
+                    newWrapMetadata);
+
+                return wrapResultMono.flatMap(
+                    wrapResult -> {
+                        byte[] wrappedDek = wrapResult.getLeft();
+                        EncryptionKeyWrapMetadata updatedMetadata = wrapResult.getMiddle();
+                        InMemoryRawDek updatedRawDek = wrapResult.getRight();
+
+                        CosmosItemRequestOptions effectiveRequestOptions = requestOptions != null ? requestOptions : new CosmosItemRequestOptions();
+
+                        effectiveRequestOptions.setIfMatchETag(dekProperties.eTag);
```

Comment:
You are modifying the passed in request options here. Wouldn't it be better to create a deep copy instead?

Reply:
addressed. good point. thank you.